url.bs

```diff
@@ -1130,6 +1130,37 @@ different document encoding. Using the <a>UTF-8</a> encoding everywhere solves t
  <li><p>Return <var>url</var>.
 </ol>
 
+<div class=example id=example-url-parsing>
+ <table>
+  <tr>
+   <th>Input
+   <th>Base
+   <th>Output
+  <tr>
+   <td><code>example</code>
+   <td><code>https://example.com/demo</code>
+   <td><code>https://example.com/example</code>
+  <tr>
+   <td><code>hello:world</code>
+   <td><code>https://example.com</code>
+   <td><code>hello:world</code>
+  <tr>
+   <td><code>https://example.com:demo</code>
+   <td>
+   <td>Failure
+  <tr>
+   <td><code>https:example.invalid</code>
```

Comment:
Reminder to self: we should add this example but against about:blank since I am pretty sure (will check) that it results in `https://example.invalid/`.